### PR TITLE
feat: add cw721-fixed-price contract

### PR DESCRIPTION
**Pull request by vernonjohnson**
This PR includes a contract for implementing fixed-price limited edition NFTs, which can be purchased with cw20 tokens.

### CODE REVIEW

**Review by JakeHartnell — DISMISSED** [review]
Very nice! Great to have an example like this. Only thing I'm not 100% sold on is the name for this contract... because it's not actually a `cw721` but rather a contract that uses one.

Curious what others think, but otherwise LGTM.

**Comment by vernonjohnson** [comment]
@JakeHartnell Good point. I can rename it to just "editions".

**Comment by orkunkl** [comment]
I think using `cw721-x` naming is convenient for extensions, or maybe `cw721-ext-x` @ethanfrey @JakeHartnell wdyt ?
@verminator23 renaming the project to `cw721-fixed-price` makes sense.

**Review by orkunkl — CHANGES_REQUESTED** [review]
We should resolve project naming.

**Comment by ethanfrey** [comment]
This looks like a very useful functionality (cw721 minting contract) that many people have asked for. They can either use this as is, or as inspiration to customize further. It think this is a valuable addition.

About the naming, we use `cw20-` for a number of related contracts, not just `cw20-staking`, which implements `cw20` interface, but also `cw20-escrow` and `cw20-ics20` which accept cw20 tokens and do something with them. As long as it is clear in the README, I think this is fine.

I haven't done a code review (just a brief skim), but I support adding it here (after someone does a better review)

**Comment by ethanfrey** [comment]
`cw721-editions`
`cw721-minter`
`cw721-fixed-price`
`cw721-so-this-is-how-you-let-people-mint-nfts-by-paying`

are all fine with me. Naming is tough. But let's not kick ourselves too much and just pick something that is "good enough"

**Comment by vernonjohnson** [comment]
Appreciate the comments from everyone. Will rename it to "cw721-fixed-price" later today.

**Comment by vernonjohnson** [comment]
Changed the name of the contract to cw721-fixed-price. Lmk if there's anything else you guys think could be improved.

**Comment by the-frey** [comment]
Code looks good, but some way of expressing it's paid for in CW20 in the naming (hard I know 🙃) would be my only nitpick

`cw721-mint-with-cw20`?
`cw721-mint-paid-with-cw20`?

Ew. Actually:

`cw20-mints-cw721` is probably the shortest I can think of

**Review by orkunkl — DISMISSED** [review]

File: contracts/cw721-fixed-price/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  cosmwasm/rust-optimizer:0.12.4
+"""
+
+[dependencies]
+cosmwasm-std = { version = "1.0.0-beta" }
+cosmwasm-storage = { version = "1.0.0-beta" }
+cw-storage-plus = "0.10"
+cw2 = "0.10"
+schemars = "0.8.3"
+cw721-base = { path = "../cw721-base", version = "0.10.1" }
```
Comment (orkunkl) [inline contracts/cw721-fixed-price/Cargo.toml]:
```suggestion
cw721-base = { path = "../cw721-base", version = "0.10.1" features = [ "library"] }
```

Comment (orkunkl) [reply contracts/cw721-fixed-price/Cargo.toml]: This will reduce the code size

**Review by JakeHartnell — DISMISSED** [review]
LGTM, nice to have an example like this.

**Review by orkunkl — DISMISSED** [review]
LGTM

**Review by JakeHartnell — DISMISSED** [review]
LGTM

File: contracts/cw721-fixed-price/examples/schema.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+fn main() {
+    let mut out_dir = current_dir().unwrap();
+    out_dir.push("schema");
+    create_dir_all(&out_dir).unwrap();
+    remove_schemas(&out_dir).unwrap();
+
+    export_schema(&schema_for!(InstantiateMsg), &out_dir);
+    export_schema(&schema_for!(ExecuteMsg), &out_dir);
+    export_schema(&schema_for!(QueryMsg), &out_dir);
```
Comment (orkunkl) [inline contracts/cw721-fixed-price/examples/schema.rs:16]: Best to add `Config` here too

**Review by orkunkl — DISMISSED** [review]

**Review by orkunkl — APPROVED** [review]
🚀

**Pull request merged**